docs(general): provide oop example that matches cli output and fix functional example

thomas-w-grant opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Documentation for the repository pattern has the current behavior:
1.  Functional example calls `createStore` with `{name}` with no name set.  
2. OOP acronym is not defined.  Learner bias that everyone knows that OOP is Object-oriented programming.  Most do, but such little effort to serve all.  ❤️ 
3. OOP example is not complete:  Instead, you get `// ...`.  Approach bias.  Functional fans get a complete example.  OOP fans do not.    Not friendly to your Angular friends.
4. OOP example does not match the awesome output produced by the CLI when `repoTemplate` option is set to `class` for `repo` command.

Issue Number: N/A

## What is the new behavior?

Documentation for the repository pattern now has the following behavior:
1. Functional example calls `createStore` with `{name: 'auth'}`.
2. OOP acronym is defined once with the following text: "object-oriented programming (OOP)"
3. OOP example is equally complete as the functional example.
4. OOP example now matches the awesome output produced by the CLI.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

NetanelBasal commented on docs/docs/repository.mdx:
```diff
@@ -26,21 +29,45 @@ The Repository pattern provides 2 main benefits:
 2. It encourages you to implement all store operations in one place, making your code more reusable and easy to find.
 
-You can also use the `OOP` approach:
+If working in Angular, you can also use the object-oriented programming (OOP) approach:
```
This shouldn't be specific to Angular.

thomas-w-grant replied:
👍🏻 I definitely second guessed myself when I wrote that.  I will remove it.

I based my decision to include it on seeing this in the CLI documentation:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/76440551/168470717-55c9d6ba-2d92-4c7a-9161-3e75c1c2460b.png">

NetanelBasal commented on docs/docs/repository.mdx:
```diff
@@ -26,21 +29,45 @@ The Repository pattern provides 2 main benefits:
+    });
+  }
+
+  private createStore(): typeof store {
```
This style will be used only if you set the [inlineStoreInClass](https://ngneat.github.io/elf/docs/cli#config) to `true`.

NetanelBasal replied:
We can add a new block showing this example.

thomas-w-grant replied:
You got me thinking.  Should these new examples go in the repository documentation or the CLI documentation.  I could make an argument for both.  What are your thoughts?

Some feedback, the team that I work on is considering using Elf but was very concerned about the lack of Angular examples.  I told them that the CLI produces really great Angular code (in my opinion (IMO)).  Their feedback was that they wished the documentation showed it.

NetanelBasal replied:
We can open a dedicated section of "Angular Examples"

thomas-w-grant replied:
Oh .... I like that.  ❤️

thomas-w-grant commented:
I have removed the mention of Angular and I have removed the example.  Perhaps it will appear in the near future in a "Angular examples" section.  😄 🙏🏻 ❤️
